### ISSUE TITLE
[InfoFromDev]Wheel loader and similar tools Issue collection.

### ISSUE THREAD

**Issue by Tensuko** [Courseplay/Courseplay_FS22#2783]
As said in the changelog, we were not able to test all different kinds of combinations.
In case you have a tool that does weird things and is not working correctly, pls post it here with a screenshot of the situation (you can post screenshots by drag and drop them in the comment box, no need to upload them somewhere else.)

**Comment by Dusixxxxx** [comment]
![1](https://github.com/Courseplay/Courseplay_FS22/assets/145778250/14d72e49-1925-4a67-ad63-d4905dd5ec91)
Hello, I noticed that for the CP to work, the vehicle to be loaded must be in the red circle area. On many maps this is impossible due to lack of space. Can the charging area be extended (the red circle in the picture)
PS Tried with several cards.
Thank you for the most useful MOD

**Comment by Tensuko** [comment]
Got an Idea to solve this, just wait for us finding a good solution :)

**Comment by Tensuko** [comment]
Placable search area (size changed a bit).
![grafik](https://github.com/Courseplay/Courseplay_FS22/assets/27344744/765ccddd-ed37-4bc3-b870-4790639be0eb)

**Comment by ct177mak** [comment]
How far away from the silo can the Placable search area be?

**Comment by Dusixxxxx** [comment]
Hello, the idea is good, but it doesn't work. The problem was not solved. If you want a log file, I'm available.
Good work ;)

**Comment by Tensuko** [comment]
> How far away from the silo can the Placable search area be?

180m

**Comment by Tensuko** [comment]
> Hello, the idea is good, but it doesn't work. The problem was not solved. If you want a log file, I'm available. Good work ;)

And what did you do to try it out?

**Comment by Tensuko** [comment]
Now it should work.

**Comment by JDTiger** [comment]
Hey, the Volvo L180H with the high tip bucket pushes the bucket too low/tips it too far forward for loading.
the bucket scrapes so low over the ground, that the loader can't steer anymore

https://youtu.be/3UCCMtyXhIU

**Comment by blubstar86** [comment]
Hes doing one round and then he stops with "waiting for driver"
In the silo is compost from this mod [https://www.farming-simulator.com/mod.php?lang=de&country=de&mod_id=274545&title=fs2022](url)

[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/12872937/log.txt)

![image](https://github.com/Courseplay/Courseplay_FS22/assets/47928255/7bb5c49d-45ab-4dc4-80f5-30a47c072bce)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/47928255/6a789d0f-be97-4bfa-b6f2-442f0efa0e7b)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/47928255/0bcde47b-2208-4a74-b518-a4f1180842d8)

**Comment by Tensuko** [comment]
Is compost in the trailer ? Log says not valid for loading, thats why he is waiting for one.

**Comment by blubstar86** [comment]
Yes it is, see last screenshot.

And its very strange, everytime i stop and start the loader, he does his job one time and then stands there and refuses to work :)

Now i have this combo and it works better, but sometimes he still stops his work
![image](https://github.com/Courseplay/Courseplay_FS22/assets/47928255/2b3ef0e0-9df9-44cf-8fab-9a77699fb52f)

**Comment by Tensuko** [comment]
Stops for the same reason?

**Comment by blubstar86** [comment]
My bad, havent checkt it and cant check it, silo is already empty. At least the info panel shows same message, but maybe the message is just wrong.

Maybe you can see something in the video i recorded yesterday
[https://drive.google.com/file/d/1wi7HvfYBZbxjcifZaDxr6OynMPcU0mnT/view?usp=sharing](url)

**Comment by BippyMiester** [comment]
So I guess I'm a little bit confused on which implements to use with this mode. I know that the telehandler with the bucket attachment does work in this mode, however a wheel loader with a bucket attachment does not. It only has the loading mode.

To reduce my confusion, is there a list of attachments that work for "Bunker Silo Mode" and not "Bunker Loading"?

**Comment by Tensuko** [comment]
Bunker Silo Mode should be compacting wich means the shovel you attached wasn't noticed.
Wich one is it? Because all I used where noticed correctly.

There is no list as there are endless options and especially mods where the modder put it wrong specs might not work where we have no influence.

**Comment by Tensuko** [comment]
Wheelloader with buckets are not for compacting and leveling....

**Comment by simonIOW** [comment]
<img width="1280" alt="Screenshot 2023-10-17 203052" src="https://github.com/Courseplay/Courseplay_FS22/assets/39086148/76c9625e-5819-4063-a6ea-df6fd0593dc9">
<img width="1280" alt="Screenshot 2023-10-17 203740" src="https://github.com/Courseplay/Courseplay_FS22/assets/39086148/baaf2299-9ecf-427b-ade0-e09034289295">

Volvo L180H + MES400 Leveler wanders off when an unloader arrives, rather than waiting at park node.  If it hits something, it will spin its wheels, else it reports as off course.  Where it ends up is random.

**Comment by Tensuko** [comment]
@schwiti6190 is the fix for the volvo wheelloader only applied to silo loading or global?

**Comment by Crinouche** [comment]
Hello Schwiti6190,

Thanks for your feedback.
Yellow is from the base game
Here are the others:
https://www.farming-simulator.com/mod.php?lang=fr&country=fr&mod_id=242983&title=fs2022
https://www.farming-simulator.com/mod.php?lang=fr&country=fr&mod_id=240707&title=fs2022
https://www.farming-simulator.com/mod.php?lang=fr&country=fr&mod_id=246524&title=fs2022
I tested the last one with a tractor and a telehandler and it doesn't work either.
The telehandler cannot even empty into the trailer (the arm would also have to extend to reach the trailer)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/36656729/e87ab23d-4882-4514-94fa-4f937c5eafbf)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/36656729/8c6ce902-9d4b-463d-965e-5b815b4a4d90)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/36656729/52818c15-3755-4b58-a78e-6b383584e20a)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/36656729/5d41a072-57e6-44e3-a6c6-f1821044f4c0)

**Comment by schwiti6190** [comment]
@Crinouche Hopefully I now got them all configured.
You can check out this PR: https://github.com/Courseplay/Courseplay_FS22/pull/2867 if you like and give it another try.
I will probably implement the telelader extension but that might take a long time until I get around to implemnt it.

**Comment by Crinouche** [comment]
Hello Schwiti6190,
It works now. Thanks.
For Telescopics, I can wait. :)

However, I have another problem. The buckets do not collect down enough and leave a layer of silage on the ground. But here, I don't think it comes from Courseplay.
I will do more tests

**Comment by Tensuko** [comment]
For that there is the shovel hight adjust on the hud.

**Comment by Crinouche** [comment]
> Pour cela il y a le réglage de la hauteur de la pelle sur le hud.

I know that. The problem is elsewhere

**Comment by ct177mak** [comment]
@Crinouche There is also a fashion scoop that helps to pick up anything that is uneven. Shovels with terrain compensation so that nothing remains in depressions. https://forbidden-mods.de/filebase/file/34825-schaufeln-f%C3%BCr-front-rad-und-telelader/

**Comment by simonIOW** [comment]
Fix for silo bunker driver when leveling is working, thank you.

One more with the Volvo L180H.  When shoveling it gets blocked on the side of the silo when exiting for the unloading point.  

<img width="1280" alt="Screenshot 2023-10-20 221147" src="https://github.com/Courseplay/Courseplay_FS22/assets/39086148/6d5e06ca-1528-4d31-8697-7ee2d589a2af">

This is the case regardless of how wide the work width is set (it takes the work width into account and still behaves as if blocked)

**Comment by Tensuko** [comment]
Hmm thats a still too small turn radius.
Need to figure something out.

**Comment by simonIOW** [comment]
> Hmm thats a still too small turn radius. Need to figure something out.
> 
> Oh no I remember. No matter how high the radius gets, the Volvo wont make it. The problem is that we had to kind of hack the volvo to work because Giants build them wrong ( its a bit complicated...)

I'll swap to another loader 😂

**Comment by simonIOW** [comment]
@Tensuko Question/thought, is there a way to get the volvo/loader to back up so it is at a right angle to the silo when blocked, and then try the radius again?  That way it would eventually get around?

<img width="1280" alt="Screenshot 2023-10-20 221147" src="https://github.com/Courseplay/Courseplay_FS22/assets/39086148/0006cd52-9216-498a-82e0-2ba72c7e90fc">

Back to the Torion for me in the interim :)

**Comment by Tensuko** [comment]
I have a different Idea that should solve general blocking issues. But need to see if that will work.

**Comment by simonIOW** [comment]
Found another issue with the silo driver when leveling.  Different bunker, and the error message is different, but the result is the same (driver keeps reversing)

<img width="1280" alt="Screenshot 2023-10-22 170015" src="https://github.com/Courseplay/Courseplay_FS22/assets/39086148/c7422931-a5b8-40d7-9cfc-b6eabdb46a57">

On the first bunker the fix works fine 🤷‍♂️

**Comment by Tensuko** [comment]
As I said...
I have a different Idea that should solve general blocking issues.
Just wait for that....

**Comment by blubstar86** [comment]
Update:

> Stops for the same reason? (In the log)

**1. Test:**
- Filled the Silo almost completely
- started the job with the robo-six
![image](https://github.com/Courseplay/Courseplay_FS22/assets/47928255/130b9250-d05f-4bde-9237-c6718c3d34c5)
- OMFG he worked for like 8h+ without any issue with his tiny shovel and i loved it, thanks a lot for that mode!
- but in the end when silo is almost empty he refused to work like in my first post. (wayting for driver)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/47928255/36c140a7-b1b6-4c14-b378-66aa95046035)
- then i thought, maybe he is recognizing some stuff that fell over the silo and isnt "Compost" (Some parts where straw and biomass)
![image](https://github.com/Courseplay/Courseplay_FS22/assets/47928255/2dbf98f9-e157-480b-a5bb-654760941cd0)
- so maybe he just thinks he doesnt have compost in his shovel and then  he cant find a driver?


**2. Test:**
- very flat filled silo, but this time with 100% nothing else arround, no straw, no biomass that fell over the silo. just to make sure no other filltypes messing arround.
- started the job with the robo-six
- worked without issue till last row
https://drive.google.com/file/d/1QUzuH338bNmjTKAvj_mgDS2oSK9s5VKe/view?usp=sharing
- he doesent drive far enough to reach the last leftover and then he stops with "waiting for driver"
- switched to New Holland W190D
https://drive.google.com/file/d/124UVfpZ7CtDwfOP5mc7cx3o2j5DpmjOh/view?usp=sharing
- he was able to drive far enough to catch the leftover and cleand the silo to 100%
- the one last try and he stops with same message "waiting for driver"
- could it be that it is a "im done with my task, cant reach anything" message?


Sorry for the wall of text, but maybe it can help to do some improvements.

Like i said earlier, thanks a lot for the 8h+ work of the robo-six without any issues!

**Comment by Tensuko** [comment]
The last tiny bit of the Bunker can't be loaded without risking crashing into the silo wall or get too far outside the silo.
We tried to fix this but it won't work without crashes.
I think the last tiny bit can be loaded by player, when you have loaded a few hundred thousand or even a dew million with the CP Driver :)

This Issue will be closed now as we think we have the driver now in a very good state.
Thanks all for your feedback.
The last fixes and adjustments for now will come with 7.4.0.1
